File: 20190519python/0389. Find the Difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        s_dict={}
        for c in s:
            if c not in s_dict:
                s_dict[c]=1
            else:
                s_dict[c]+=1
        for c in t:
            if c not in s_dict or s_dict[c]==0:
                return c
            s_dict[c]-=1

File: archives/C#/0389. Find the Difference.cs
public class Solution {
    public char FindTheDifference(string s, string t) {
        Dictionary<char,int> tDict=new Dictionary<char,int>();
        foreach(char c in t){
            if(!tDict.ContainsKey(c)){
                tDict[c]=1;
            }
            else{
                tDict[c]+=1;
            }
        }
        foreach(char c in s){
            tDict[c]-=1;
        }
        char rep=new char();
        foreach(KeyValuePair<char,int> kvp in tDict){
            if (kvp.Value!=0){
                rep=kvp.Key;
            }
        }
        return rep;
        
    }
}
